tools/test-rules.sh

```diff
@@ -5,7 +5,7 @@ count=0
 while true; do
 	echo "running test ..."
 	go clean -testcache
-	if ! go test -p=1 -race ./client -v -run TestRules; then
+	if ! go test -p=1 -race ../client -v -run TestRules; then
```

Comment:
I guess this is personal preference, but I typically run that script from the top level project directory:

`./tools/test-rules.sh`

There is no need to cd into the tools dir first.

Again, this is personal preference, but it would probably be good to understand why things are they way they are before changing them.